bcda/auth/client/ssas_test.go

```diff
@@ -465,3 +487,265 @@ func (s *SSASClientTestSuite) TestGetToken() {
 func TestSSASClientTestSuite(t *testing.T) {
 	suite.Run(t, new(SSASClientTestSuite))
 }
+
+func (s *SSASClientTestSuite) TestPingTable() {
+	tests := []struct {
+		fnInput       []string
+		env           EnvVars
+		errorExpected bool
+		message       string
+	}{
+		{fnInput: []string{}, env: EnvVars{}, errorExpected: false},
+		{fnInput: []string{}, env: EnvVars{BCDA_SSAS_CLIENT_ID: "fake"}, errorExpected: true, message: "introspect request failed; 401"},
+		{fnInput: []string{}, env: EnvVars{SSAS_PUBLIC_URL: "\n"}, errorExpected: true, message: "bad request structure: parse \"\\n/introspect\": net/url: invalid control character in URL"},
+		{fnInput: []string{}, env: EnvVars{SSAS_PUBLIC_URL: "localhost"}, errorExpected: true, message: "introspect request failed: Post \"localhost/introspect\": unsupported protocol scheme \"\""},
+		{fnInput: []string{}, env: EnvVars{BCDA_SSAS_CLIENT_ID: "-1"}, errorExpected: true, message: "missing clientID or secret"},
+	}
+
+	for _, tc := range tests {
+
+		s.setEnvVars(tc.env)
+		client, err := authclient.NewSSASClient()
+
+		if err != nil {
+			s.FailNow(constants.CreateSsasErr, err.Error())
+		}
+		err = client.Ping()
+
+		if tc.errorExpected {
+			assert.EqualError(s.T(), err, tc.message)
+		} else {
+			assert.Nil(s.T(), err)
+		}
+
+	}
+}
+
+func (s *SSASClientTestSuite) TestGetPublicKeyTable() {
+	tests := []struct {
+		fnInput       []string
+		header        int
+		env           EnvVars
+		errorExpected bool
+		message       string
+	}{
+		{fnInput: []string{}, header: http.StatusOK, env: EnvVars{}, errorExpected: false, message: ""},
+		{fnInput: []string{}, header: http.StatusOK, env: EnvVars{SSAS_URL: "\n"}, errorExpected: true, message: "could not get public key: parse \"\\n/system/1/key\": net/url: invalid control character in URL"},
+		{fnInput: []string{}, header: http.StatusOK, env: EnvVars{SSAS_URL: "localhost"}, errorExpected: true, message: "could not get public key: Get \"localhost/system/1/key\": unsupported protocol scheme \"\""},
+		{fnInput: []string{}, header: http.StatusOK, env: EnvVars{BCDA_SSAS_CLIENT_ID: "-1"}, errorExpected: true, message: "missing clientID or secret"},
+	}
+
+	for _, tc := range tests {
+		router := chi.NewRouter()
+		keyStr := "123456"
+		router.Get("/system/{systemID}/key", func(w http.ResponseWriter, r *http.Request) {
+			_, err := w.Write([]byte(`{ "client_id": "123456", "public_key": "` + keyStr + `" }`))
+			if err != nil {
+				log.Fatal(err)
```

Comment:
should any errors that occur here result in a test failure instead of logging and exiting?

Reply:
On this one, I think I have a conceptual question: In the context of a test, would we behave any differently for one vs. the other?

In this case, we're mocking the router to return a fake public key. An error here would imply that the router failed to write a string. If the response is written incorrectly/errors out (and we don't log + exit), the test that relies upon that response would have an unexpected outcome compared to what we're asserting, and the test would then fail. 
Thinking further, my personal thoughts would be that if we're setting up the testing conditions incorrectly, we shouldn't necessarily trust the validity of the test result, so indeed it may make sense to exit where the setup failed? I can see it both ways, not particularly attached to one approach over the other, and interested to get your perspective.

Reply:
That's a really good point! I think maybe we should leave that in, then. The `testing` package offers a logger that will use the same behavior but continue test execution: https://pkg.go.dev/testing#T.Fatal

Maybe that would be an option to use `t.Fail(err)`? What are your thoughts?

Reply:
Sounds good, thank you! I've updated that + removed the fmt.println statements in a new commit that should go through the c/i flow shortly.

Reply:
perfect, thanks for the update!